core/unix/os_exports.h

```diff
@@ -112,6 +115,17 @@
 # error NYI
 #endif
 
+#if defined(MACOS) && defined(X64)
+/* FIXME: current pthread_t struct has the first TLS entry at offset 28. We should
+ * provide a dynamic method to determine the first entry for forward compatability.
+ * Starting w/ libpthread-218.1.3 they now leave slots 6 and 11 unused to allow
+ * limited interoperability w/ code targeting the Windows x64 ABI. We steal slot 6
+ * for our own use.
```

Comment:
I think a decision should be made before progressing any further. For option #2, we will have a problem w/ the main thread's TLS, since it is located in the bss segment. We won't be able to add an extra page to the left w/ 16-bit offs (unless I'm misunderstanding that approach?). Continuing w/ the approach in this commit, we can go for option #3 or #6. Additionally, with these issues, would it be more worth while to focus efforts on the private loader instead? WDYT?